pkgs/applications/version-management/rapidsvn/default.nix

```diff
@@ -9,19 +9,36 @@ stdenv.mkDerivation rec {
     sha256 = "1bmcqjc12k5w0z40k7fkk8iysqv4fw33i80gvcmbakby3d4d4i4p";
   };
 
-  buildInputs = [ wxGTK subversion apr aprutil python2 ];
+  buildInputs = [ wxGTK subversion apr aprutil python3 ];
+
+  NIX_CFLAGS_COMPILE = [ "-std=c++14" ];
 
   configureFlags = [ "--with-svn-include=${subversion.dev}/include"
     "--with-svn-lib=${subversion.out}/lib" ];
 
   patches = [
     ./fix-build.patch
+    # Python 3 compatibility patches
+    (fetchpatch {
+      url = "https://github.com/RapidSVN/RapidSVN/pull/44/commits/2e26fd5d6a413d6c3a055c17ac4840b95d1537e9.patch";
+      hash = "sha256-8acABzscgZh1bfAt35KHfU+nfaiO7P1b+lh34Bj0REI=";
+    })
+    (fetchpatch {
+      url = "https://github.com/RapidSVN/RapidSVN/pull/44/commits/92927af764f92b3731333ed3dba637f98611167a.patch";
+      hash = "sha256-4PdShGcfFwxjdI3ygbnKFAa8l9dGERq/xSl54WisgKM=";
+    })
+    (fetchpatch {
+      url = "https://github.com/RapidSVN/RapidSVN/pull/44/commits/3e375f11d94cb8faddb8b7417354a9fb51f304ec.patch";
+      hash = "sha256-BUpCMEH7jctOLtJktDUE52bxexfLemLItZ0IgdAnq9g=";
+    })
```

Comment:
Sorry for asking this, but could we maybe bump this package to build from `master` instead? This 0.12.1 release is ancient (from 2012!) and all those patches were merged. It also seems that the `master` includes some fixes that may make this package works with newer versions of C++ (so maybe the `NIX_CFLAGS_COMPILE` will become unnecessary).

Reply:
I have no preference either way (I do not use the package myself, I just saw it was broken). I took a quick look at what other distributions do and it looks like Fedora is using `master`. I will try to do the update to see what happen.

Reply:
So after a quick test it appears that even if we use `master` (https://github.com/RapidSVN/RapidSVN/commit/3a564e071c3c792f5d733a9433b9765031f8eed0) we will still need to keep `NIX_CFLAGS_COMPILE` and we also get another bunch of new issues to deal with. I'm not overly confident I can manage the upgrade without introducing some regressions.